### PR TITLE
AIP-65: Add back DAG versioning support

### PR DESCRIPTION
**Pull request by ephraimbuddy**
This reverts commit b757bd8df824d4eba952f6e140bcb373bc3f1003.

### CODE REVIEW

**Comment by potiuk** [comment]
Yeah.. Looks like fairly `reliably flaky` (if we can use such oxymoron :D ).

File: tests/jobs/test_scheduler_job.py [hunk 1]
```diff
@@ -116,14 +116,6 @@ def disable_load_example():
             yield
 
 
-@pytest.fixture(scope="module")
```
Comment (potiuk) [inline tests/jobs/test_scheduler_job.py:118]: HA!

**Review by potiuk — APPROVED** [review]
Should be good, though i'd be more happy if this one is rebased (and tests are green) after https://github.com/apache/airflow/pull/43771 that should fix google.auth pickiness about mocks used.

**Comment by ephraimbuddy** [comment]
> Should be good, though i'd be more happy if this one is rebased (and tests are green) after #43771 that should fix google.auth pickiness about mocks used.

Yeah, rebased, and it was green except for a Kubernetes test and task sdk test that stopped midway. Have rebased again. Let's see how it goes

**Pull request merged**